src/base/Roles/CrossChain/CrossChainOPTellerWithMultiAssetSupport.sol

```diff
@@ -0,0 +1,80 @@
+
+
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import {CrossChainTellerBase, BridgeData, ERC20} from "./CrossChainTellerBase.sol";
+
+interface ICrossDomainMessenger {
+    function xDomainMessageSender() external view returns (address);
+    function sendMessage(
+        address _target,
+        bytes calldata _message,
+        uint32 _gasLimit
+    ) external;
+}
+
+/**
+ * @title CrossChainLayerZeroTellerWithMultiAssetSupport
+ * @notice LayerZero implementation of CrossChainTeller 
+ */
+contract CrossChainOPTellerWithMultiAssetSupport is CrossChainTellerBase {
+
+    ICrossDomainMessenger public messenger;
+
+    constructor(address _owner, address _vault, address _accountant, address _weth, address _messenger)
+        CrossChainTellerBase(_owner, _vault, _accountant, _weth)
+    {
+        messenger = ICrossDomainMessenger(_messenger);
+    }
+
+    /**
+     * @dev the virtual bridge function to be overridden
+     * @param data bridge data
+     * @return messageId
+     */
+    function _bridge(uint256 shareAmount, BridgeData calldata data) internal override returns(bytes32){
+        messenger.sendMessage(
+            address(selectorToChains[data.chainSelector].targetTeller),
+            abi.encodeCall(
+                this.receiveBridgeMessage,
+                (
+                    data.destinationChainReceiver,
+                    shareAmount
+                )
+            ),
+            uint32(data.messageGas)
+        );
+    }
+
+    function receiveBridgeMessage(address receiver, uint256 shareMintAmount) external{
+        // TODO
+        // Clean this up
+        require(
+            msg.sender == address(messenger),
+            "Greeter: Direct sender must be the CrossDomainMessenger"
+        );
```

Comment:
Let's standardize to custom errors and not require statements?

Reply:
Will do, this was copied straight from the OP example